solver/src/http_solver.rs

```diff
@@ -114,140 +155,111 @@ impl HttpSolver {
+            status
+        );
+        serde_json::from_str(body_str).context("failed to decode response json")
     }
 }
 
-// types used in the solver http api
-
-#[derive(Debug, Default, Serialize)]
-struct BatchAuctionModel {
-    tokens: HashMap<String, TokenInfoModel>,
-    orders: HashMap<String, OrderModel>,
-    uniswaps: HashMap<String, UniswapModel>,
-    ref_token: String,
-    #[serde(serialize_with = "serialize_as_string")]
-    default_fee: f32,
-}
-
-#[derive(Debug, Deserialize)]
-struct Solution {
-    // TODO: wait for solution format to be documented
-}
-
-#[derive(Debug, Serialize)]
-struct OrderModel {
-    sell_token: String,
-    buy_token: String,
-    #[serde(with = "u256_decimal")]
-    sell_amount: U256,
-    #[serde(with = "u256_decimal")]
-    buy_amount: U256,
-    allow_partial_fill: bool,
-    is_sell_order: bool,
-}
-
-#[derive(Debug, Serialize)]
-struct UniswapModel {
-    token1: String,
-    token2: String,
-    #[serde(serialize_with = "serialize_as_string")]
-    balance1: u128,
-    #[serde(serialize_with = "serialize_as_string")]
-    balance2: u128,
-    #[serde(serialize_with = "serialize_as_string")]
-    fee: f32,
-    mandatory: bool,
-}
-
-#[derive(Debug, Serialize)]
-struct TokenInfoModel {
-    #[serde(serialize_with = "serialize_as_string")]
-    decimals: u32,
-}
-
-fn serialize_as_string<S>(t: &impl ToString, serializer: S) -> Result<S::Ok, S::Error>
-where
-    S: Serializer,
-{
-    serializer.serialize_str(t.to_string().as_str())
+#[async_trait::async_trait]
+impl Solver for HttpSolver {
+    async fn solve(&self, liquidity: Vec<Liquidity>) -> Result<Option<Settlement>> {
+        let tokens = self.tokens(liquidity.as_slice());
+        let orders = self.orders(liquidity.as_slice());
+        let amms = self.amms(liquidity.as_slice());
+        let ref_token = match tokens.keys().next() {
+            Some(token) => token.clone(),
+            None => return Ok(None),
+        };
+        let model = BatchAuctionModel {
+            tokens: self.token_models(&tokens),
+            orders: self.order_models(&orders),
+            uniswaps: self.amm_models(&amms),
+            ref_token,
+            default_fee: 0.0,
+        };
+        let settled = self.send(&model).await?;
+        tracing::debug!("optimizer response {:?}", settled);
+        settlement::convert_settlement(&settled, &tokens, &orders, &amms).map(Some)
+    }
 }
 
 #[cfg(test)]
 mod tests {
-    use crate::liquidity::{LimitOrder, MockLimitOrderSettlementHandling};
-    use std::sync::Arc;
-
     use super::*;
+    use crate::liquidity::{
+        AmmOrder, LimitOrder, MockAmmSettlementHandling, MockLimitOrderSettlementHandling,
+    };
+    use ::model::TokenPair;
+    use num::Rational;
+    use std::sync::Arc;
 
     // cargo test real_solver -- --ignored --nocapture
+    // set the env variable GP_V2_OPTIMIZER_URL to use a non localhost optimizer
     #[tokio::test]
```

Comment:
I am using an env variable because it contains credentials which we don't want in version control.